src/hooks/useSmartContract.tsx

```diff
@@ -97,48 +45,86 @@ export const useSmartContract = () => {
 
   const executeFunctionWithUProvider = useCallback(async () => {
     try {
-      if (!client || !walletConnected) {
+      if (!client) {
         console.log("Client or wallet is not connected.");
         return;
       }
-      const amount: number = 1; // Example amount for minting
       const contractAddress: `0x${string}` =
-        "0x86d112121996f767ab50bc9aad23c5454b72b739"; // Example of smart contract LSP7 address already deployed
+        "0x4E1Fe6B4085D79F5F500B835f3a2a56F27994338"; // Example of a custom smart contract LSP7 address already deployed
 
       console.log("Initializing provider and signer...");
-      const provider = await getProvider();
-      const signer = await provider.getSigner();
 
       console.log("Creating contract instance...");
-      const contract = await getContractInstance(contractAddress, signer);
-
-      const mintArgs: [string, bigint, boolean, string] = [
-        contextAccounts[0] as `0x${string}`,
-        ethers.parseUnits(amount.toString(), "wei"), // Amount could be ethers or wei
-        false as boolean,
-        "0x" as string,
-      ];
+      const contract = await getContractInstance(contractAddress, client);
+      console.log("contract", contract);
+      const data: string = contract.interface.encodeFunctionData("allCanMint"); // this is a custom function
 
-      const data: string = contract.interface.encodeFunctionData(
-        "mint",
-        mintArgs
-      );
-
-      // Sending the transaction via the grid provider
       console.log("Sending transaction...");
       const txResponse = await client.sendTransaction({
-        account: contextAccounts[0] as `0x${string}`,
+        account: accounts[0] as `0x${string}`,
         to: contractAddress as `0x${string}`,
         data: data,
       });
 
       console.log("Transaction response:", txResponse);
     } catch (error) {
       console.error("Transaction failed:", error);
-      throw error;
+      return;
```

Comment:
This is not necessary, because it's already at the end of the function. Unfortunately the txResponse is not stored anywhere so the effect's result cannot be read by the UI.

Reply:
And if it throws an error will just silently fail. Not sure if that’s a good idea